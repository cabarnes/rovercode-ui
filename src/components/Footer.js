import React from 'react';
import { hot } from 'react-hot-loader';
import {
  Grid,
  Link,
  Typography,
} from '@material-ui/core';
import { FormattedMessage } from 'react-intl';

const Footer = () => (
  <Grid container direction="column" justify="center" alignItems="center">
    <Grid item>
      <Typography variant="subtitle1">
        <FormattedMessage
          id="app.protected_route.conduct_message"
          description="Description and link to inform the user of the location of the Rovercode Code of Conduct"
          defaultMessage="All users must follow the "
        />
        <Link href="https://docs.rovercode.com/conduct/code-of-conduct">
          <FormattedMessage
            id="app.protected_route.conduct_link"
            description="Rovercode Code of Conduct"
            defaultMessage="Rovercode Code of Conduct."
          />
        </Link>
        <FormattedMessage
          id="app.protected_route.inappropriate_message"
          description="Instructions for what a student should do if they experience harassment"
          defaultMessage=" If you see inappropriate behavior or feel you are being harassed, please stop using Rovercode and tell your teacher"
        />
        <FormattedMessage
          id="app.protected_route.teachers_message"
          description="Description and link to inform the teacher of where to report issues"
          defaultMessage=" Teachers, visit "
        />
        <Link href="https://docs.rovercode.com/conduct/report">
          <FormattedMessage
            id="app.protected_route.teachers_link"
            description="Teacher conduct reporting link"
            defaultMessage="this page "
          />
        </Link>
        <FormattedMessage
          id="app.protected_route.teachers_secondmessage"
          description="Teacher conduct reporting link"
          defaultMessage="to report the issue."
        />
      </Typography>
    </Grid>
  </Grid>
);

export default hot(module)(Footer);
