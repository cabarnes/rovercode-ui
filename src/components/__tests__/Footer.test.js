import React from 'react';
import { shallow } from 'enzyme';

import Footer from '../Footer';

describe('The Footer component', () => {
  test('renders on the page with no errors', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });
});
