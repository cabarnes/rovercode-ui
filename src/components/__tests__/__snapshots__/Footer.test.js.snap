// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Footer component renders on the page with no errors 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(Typography))
      variant="subtitle1"
    >
      <FormattedMessage
        defaultMessage="All users must follow the "
        id="app.protected_route.conduct_message"
        values={Object {}}
      />
      <WithStyles(ForwardRef(Link))
        href="https://docs.rovercode.com/conduct/code-of-conduct"
      >
        <FormattedMessage
          defaultMessage="Rovercode Code of Conduct."
          id="app.protected_route.conduct_link"
          values={Object {}}
        />
      </WithStyles(ForwardRef(Link))>
      <FormattedMessage
        defaultMessage=" If you see inappropriate behavior or feel you are being harassed, please stop using Rovercode and tell your teacher"
        id="app.protected_route.inappropriate_message"
        values={Object {}}
      />
      <FormattedMessage
        defaultMessage=" Teachers, visit "
        id="app.protected_route.teachers_message"
        values={Object {}}
      />
      <WithStyles(ForwardRef(Link))
        href="https://docs.rovercode.com/conduct/report"
      >
        <FormattedMessage
          defaultMessage="this page "
          id="app.protected_route.teachers_link"
          values={Object {}}
        />
      </WithStyles(ForwardRef(Link))>
      <FormattedMessage
        defaultMessage="to report the issue."
        id="app.protected_route.teachers_secondmessage"
        values={Object {}}
      />
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
